src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnectorConfig.java

```diff
@@ -20,42 +20,61 @@
 import io.cdap.cdap.api.annotation.Description;
 import io.cdap.cdap.api.annotation.Macro;
 import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
 import io.cdap.cdap.api.data.schema.Schema;
 import io.cdap.cdap.api.plugin.PluginConfig;
 import io.cdap.cdap.etl.api.FailureCollector;
 import io.cdap.plugin.servicenow.apiclient.ServiceNowTableAPIClientImpl;
 import io.cdap.plugin.servicenow.util.ServiceNowConstants;
 import io.cdap.plugin.servicenow.util.Util;
 
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 
 /**
  * PluginConfig for ServiceNow Connector
  */
 public class ServiceNowConnectorConfig extends PluginConfig {
+  private static final String DATE_PATTERN = "yyyy-MM-dd";
+  private static final String DATE_TIME_PATTERN = "yyyy-MM-dd HH:mm:ss";
+  private static final String TIME_PATTERN = "HH:mm:ss";
+
   @Name(ServiceNowConstants.PROPERTY_CLIENT_ID)
   @Macro
+  @Nullable
```

Comment:
why are these all getting changed to nullable? They seem like they would be required?

Reply:
Client_ID is required for establishing a connection. Must have been missed out getting adding @Nullable annotation by the developer earlier. e63c214cf864e664ca24072f21c719c1a777b9c4

Reply:
There was a bug identified by the QA team when we were using the connection manager, the pipeline is not getting deployed with the error that all these properties are non-nullable.

For instance, this error comes when you try to deploy the pipeline without giving the pipeline name.